PSModules/CompositeRegistrationFunctions.psm1

```diff
@@ -0,0 +1,500 @@
+        [string] $Path
+    )
+
+    $finalUrl = "$($script:PathApiUrl)/paths/$($Path)"
+
+    return Invoke-CompositeApiRegistrationRequest -Url $finalUrl -Method Get
+}
+
+[CmdletBinding]
+function Get-RegionRegistration
+{
+<#
+.SYNOPSIS
+Get a region from the path registration by path name and region
+
+.DESCRIPTION
+Get a region from the path registration by path name and region
+
+.PARAMETER Path
+The name of the path registration
+
+.PARAMETER PageRegion
+The region identifier
+
+.EXAMPLE
+Get-RegionRegistration -Path somepath -PageRegion 1
+#> 
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $Path,
+        [Parameter(Mandatory=$true)]
+        [int] $PageRegion
+    ) 
+
+    $finalUrl = "$($script:RegionApiUrl)/paths/$($Path)/regions/$($PageRegion)"
+
+    return Invoke-CompositeApiRegistrationRequest -Url $finalUrl -Method Get
+}
+
+function New-PathRegistration
+{
+<#
+.SYNOPSIS
+Creates a new path registration
+
+.DESCRIPTION
+Creates a new path registration
+
+.PARAMETER Path
+An object containing the path to be created
+
+.EXAMPLE
+
+$pathObject = @{
+    Path = "somePath"
+    Layout = 1
+}
+
+New-PathRegistration -Path $pathObject
+#> 
+    param(
+        [Parameter(Mandatory=$true)]
+        [object] $Path
+    )
+
+    if($null -eq $Path.Path) { throw "Path not specified" }
+    if($null -eq $Path.Layout) { throw "Layout is mandatory when creating a page registration."}
+    if($null -eq $Path.IsOnline) { $Path.IsOnline = $true }
+    
+    $requestBody = @{
+        Path = $Path.Path
+        TopNavigationText = $Path.TopNavigationText
+        TopNavigationOrder = $Path.TopNavigationOrder
+        Layout = $Path.Layout
+        IsOnline = $Path.IsOnline
+        OfflineHtml = $Path.OfflineHtml
+        PhaseBannerUrl = $Path.PhaseBannerHtml
+        ExternalUrl = $Path.ExternalUrl
+        SitemapURL = $Path.SitemapUrl
+        RobotsURL = $Path.RobotsUrl
+    }
+
+    $requestBodyText = $requestBody | ConvertTo-Json
+
+    $finalUrl = "$($script:PathApiUrl)/paths"
+
+    return Invoke-CompositeApiRegistrationRequest -Url $finalUrl -Method Post -RequestBody $requestBodyText
+}
+
+function New-RegionRegistration
+{
+<#
+.SYNOPSIS
+Creates a new region registration
+
+.DESCRIPTION
+Creates a new region registration
+
+.PARAMETER Path
+The path that the region belongs to
+
+.PARAMETER Region
+An object containing the region to be created
+
+.EXAMPLE
+
+$regionObject = @{
+    PageRegion = 1
+}
+
+New-RegionRegistration -Path somePath -Region $regionObject
+#> 
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $Path,
+        [Parameter(Mandatory=$true)]
+        [object] $Region
+    )
+
+    if($null -eq $Region.PageRegion) { throw "PageRegion is not set for a region on path $Path."}
+    if($null -eq $Region.HealthCheckRequired) { 
+        $Region | Add-Member -NotePropertyName HealthCheckRequired -NotePropertyValue $true 
+    }
+
+    if($null -eq $Region.IsHealthy) { 
```

Comment:
Why is this being set to true if it's absent, that would normally indicate it's not required